internal/security/proxyauth/cryptor.go

```diff
@@ -15,12 +19,43 @@ import (
 	"github.com/edgexfoundry/go-mod-bootstrap/v4/di"
 )
 
+const (
+	defaultAESKeyFile = "/res/insecure_aes.key"
+	EnvAesKey         = "EDGEX_INSECURE_AES_KEY"
+)
+
+// make getenv and readFile overridable for unit tests
+var (
+	getenv   = os.Getenv
+	readFile = os.ReadFile
+)
+
+func loadDefaultAESKey() ([]byte, error) {
+	if envKey := getenv(EnvAesKey); envKey != "" {
+		return []byte(envKey), nil
+	}
+
+	if keyData, err := readFile(defaultAESKeyFile); err == nil {
+		key := []byte(string(keyData))
+		for len(key) > 0 && (key[len(key)-1] == '\n' || key[len(key)-1] == '\r' || key[len(key)-1] == ' ') {
+			key = key[:len(key)-1]
```

Comment:
You can use https://pkg.go.dev/bytes#TrimRight or https://pkg.go.dev/bytes#TrimSpace to robustly handle all types of whitespace.